cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/DefaultAdmin.java

```diff
@@ -182,4 +191,14 @@ public Void call() throws TopicNotFoundException, IOException {
     }
   }
 
+  @Override
+  public List<ArtifactInfo> listArtifacts() throws IOException {
+    return defaultArtifactManager.listArtifacts(namespaceId);
```

Comment:
should be using Retries to retry this operation if it fails in a retryable fashion